specutils/analysis/width.py

```diff
@@ -0,0 +1,190 @@
+import numpy as np
+from astropy.stats.funcs import gaussian_fwhm_to_sigma
+from ..spectra import SpectralRegion
+
+
+__all__ = ['sigma_full_width', 'full_width_half_max', 'equivalent_width']
+
+
+def _computation_wrapper(func, spectrum, region):
+
+    # No region, therefore whole spectrum.
+    if region is None:
+        return func(spectrum)
+
+    # Single region
+    elif isinstance(region, SpectralRegion):
+        return func(spectrum, region=region)
+
+    # List of regions
+    elif isinstance(region, list):
+        return [func(spectrum, region=reg) for reg in region]
+
+
+def sigma_full_width(spectrum, region=None):
+    """
+    Calculate the full width of the spectrum based on an approximate value of
+    sigma.  This will be calculated over the regions, if they are specified.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion` or list of `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the gaussian sigma width.
+
+    Returns
+    -------
+    full_width : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal
+    """
+    return _computation_wrapper(_compute_sigma_full_width, spectrum, region)
+
+
+def full_width_half_max(spectrum, region=None):
+    """
+    Calculate the full width half max of the spectrum.  This will be calculated
+    over the regions, if they are specified.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object overwhich the equivalent width will be calculated.
+
+    region: `~specutils.utils.SpectralRegion` or list of `~specutils.utils.SpectralRegion`
+        Region within the spectrum to calculate the FWHM value.
+
+    Returns
+    -------
+    full_width_half_max : `~astropy.units.Quantity` or list (based on region input)
+        Approximate full width of the signal at half max
+    """
+    return _computation_wrapper(_compute_full_width_half_max, spectrum, region)
+
+
+def equivalent_width(spectrum, region=None):
```

Comment:
Actually I think `equivalent_width` should *not* be in this PR.  While the name would seem to suggest otherwise, EQW is actually a measure of the *flux* of a line more than a *width*.  It's basically the flux normalized by the continuum level.  So I'd say it should instead be in its own module for flux measurements (which we will later populate with `flux` and related things).